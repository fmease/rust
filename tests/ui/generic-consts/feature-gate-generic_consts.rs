// FIXME: polish this up

pub trait Tr {
    const K<T>: i32;
    //~^ ERROR generic consts are experimental
}

const K<T>: i32 = 0;
//~^ ERROR generic consts are experimental

const E<>: i32 = 0;
//~^ ERROR generic consts are experimental

macro_rules! discard {
    ($item:item) => {}
}

discard! { const FREE<T>: () = (); }
//~^ ERROR generic consts are experimental

discard! { impl () { const ASSOC<const N: ()>: () = (); } }
//~^ ERROR generic consts are experimental

fn main() {}
