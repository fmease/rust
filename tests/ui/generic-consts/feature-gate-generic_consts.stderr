error[E0658]: generic consts are experimental
  --> $DIR/feature-gate-generic_consts.rs:4:12
   |
LL |     const K<T>: i32;
   |            ^^^
   |
   = help: add `#![feature(generic_consts)]` to the crate attributes to enable

error[E0658]: generic consts are experimental
  --> $DIR/feature-gate-generic_consts.rs:8:8
   |
LL | const K<T>: i32 = 0;
   |        ^^^
   |
   = help: add `#![feature(generic_consts)]` to the crate attributes to enable

error[E0658]: generic consts are experimental
  --> $DIR/feature-gate-generic_consts.rs:11:8
   |
LL | const E<>: i32 = 0;
   |        ^^
   |
   = help: add `#![feature(generic_consts)]` to the crate attributes to enable

error[E0658]: generic consts are experimental
  --> $DIR/feature-gate-generic_consts.rs:18:22
   |
LL | discard! { const FREE<T>: () = (); }
   |                      ^^^
   |
   = help: add `#![feature(generic_consts)]` to the crate attributes to enable

error[E0658]: generic consts are experimental
  --> $DIR/feature-gate-generic_consts.rs:21:33
   |
LL | discard! { impl () { const ASSOC<const N: ()>: () = (); } }
   |                                 ^^^^^^^^^^^^^
   |
   = help: add `#![feature(generic_consts)]` to the crate attributes to enable

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
